nixos/modules/services/web-apps/lemmy.nix

```diff
@@ -0,0 +1,142 @@
+{ lib, pkgs, config, ... }:
+with lib;
+let
+  cfg = config.services.lemmy;
+  settingsFormat = pkgs.formats.json { };
+in
+{
+  options.services.lemmy = {
+    enable = mkEnableOption "Enable lemmy server";
+
+    config = mkOption {
+      type = types.submodule {
+        freeformType = settingsFormat.type;
+        options.database = {
+          password = mkOption {
+            type = types.str;
+            description = "password to connect to postgres";
+          };
+        };
+        options.hostname = mkOption {
+          type = types.str;
+          default = null;
+          description = ''the domain name of your instance (eg "lemmy.ml")'';
```

Comment:
Opinion you're entirely free to ignore: Don't use this type of quoting on a single line unless you need to.